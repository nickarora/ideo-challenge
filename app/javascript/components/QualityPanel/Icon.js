import React from 'react'
import styled from 'styled-components'
import PropTypes from 'prop-types'
import CollaborationIcon from 'assets/images/qualityIcons/collaboration.png'
import EmpowermentIcon from 'assets/images/qualityIcons/empowerment.png'
import PurposeIcon from 'assets/images/qualityIcons/purpose.png'

const icons = {
  collaboration: CollaborationIcon,
  empowerment: EmpowermentIcon,
  purpose: PurposeIcon,
}

const Container = styled.section`
  height: 125px;
  width: 125px;
  margin: 0 auto;
  text-align: center;
  background-size: 100%;
`

const Icon = ({ quality }) => (
  <Container
    style={{
      backgroundImage: `url(${icons[quality.name.toLowerCase()]})`,
    }}
  />
)

Icon.propTypes = {
  quality: PropTypes.shape({
    name: PropTypes.string.isRequired,
  }).isRequired,
}

export default Icon
