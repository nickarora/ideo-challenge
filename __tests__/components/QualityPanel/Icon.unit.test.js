import Icon from '~/components/QualityPanel/Icon'

jest.mock(
  'assets/images/qualityIcons/collaboration.png',
  () => 'collaboration-img'
)
jest.mock('assets/images/qualityIcons/empowerment.png', () => 'empowerment-img')
jest.mock('assets/images/qualityIcons/purpose.png', () => 'purpose-img')

describe('Icon', () => {
  it('renders the correct icon for collaboration', () => {
    const wrapper = shallow(
      <Icon
        quality={{
          name: 'Collaboration',
        }}
      >
        CONTENT
      </Icon>
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('renders the correct icon for purpose', () => {
    const wrapper = shallow(
      <Icon
        quality={{
          name: 'Purpose',
        }}
      >
        CONTENT
      </Icon>
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('renders the correct icon for empowerment', () => {
    const wrapper = shallow(
      <Icon
        quality={{
          name: 'Empowerment',
        }}
      >
        CONTENT
      </Icon>
    )
    expect(wrapper).toMatchSnapshot()
  })
})
