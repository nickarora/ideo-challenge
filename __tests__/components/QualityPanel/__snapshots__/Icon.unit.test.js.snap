// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon renders the correct icon for collaboration 1`] = `
<styled.section
  style={
    Object {
      "backgroundImage": "url(collaboration-img)",
    }
  }
/>
`;

exports[`Icon renders the correct icon for empowerment 1`] = `
<styled.section
  style={
    Object {
      "backgroundImage": "url(empowerment-img)",
    }
  }
/>
`;

exports[`Icon renders the correct icon for purpose 1`] = `
<styled.section
  style={
    Object {
      "backgroundImage": "url(purpose-img)",
    }
  }
/>
`;
